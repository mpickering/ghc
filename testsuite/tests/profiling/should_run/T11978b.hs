
import Control.Concurrent
import Control.Concurrent.MVar
import Control.Exception
import Control.Monad

main :: IO ()
main = do
    putStrLn "Start ..."
    mvar <- newMVar (0 :: Int)

    let count = 50

    forM_ [ 1 .. count ] $ const $ forkIO $ do
            threadDelay 100
            i <- takeMVar mvar
            putMVar mvar $! i + 1

    threadDelay 1000000
    end <- takeMVar mvar
    putStrLn $ "Final result " ++ show end
    assert (end == count) $ return ()
