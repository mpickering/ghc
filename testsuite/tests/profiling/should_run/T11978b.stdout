Start ...
Final result 50
